studybuddies/src/app/services/login/login.service.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable, EventEmitter } from '@angular/core';
+import { User } from '../../models/user';
+import { HttpClient } from "@angular/common/http";
+import 'rxjs';
+import { HttpHeaders } from "@angular/common/http";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LoginService {
+
+  public eventEmitter: EventEmitter<any> =  new EventEmitter();
+  public principal: User;
+
+  constructor(private http: HttpClient) { }
+
+  storeToken(user: User) {
+
+    let username = user.username;
+    let pass = user.password;
+    let basicToken = btoa(username + ":" + pass);
+
+    let headers_object = new HttpHeaders().set("Authorization", "Basic " + basicToken);
+    return this.http.get('/api/login', { headers: headers_object }).toPromise()
+      .then(token => {
+        this.getPrincipal().then(user => {
+          this.principal = user;
+          this.eventEmitter.emit(this.principal);
+        });
+      });
+  }
+
+  logout() {
+    let res = this.http.get('/api/logout').toPromise();
+    res.then(token => {
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Frontend/pullRequest?prid=7202914)
```suggestion
    res.then((token) => {
```